File: data/gpt-4o/readable_format_HPEC/cpp/file106.cpp
#include <stdio.h>
#include <omp.h>
int var[100];
int main()
{
 for (int i=0;i<100;i++)
 var[i] = 1;
#pragma omp target map(tofrom:var) device(0)
 {
#pragma omp parallel for ordered
 for (int i = 1; i < 100; i++)
 {
 #pragma omp ordered
 var[i] = var[i-1]+1;
 }
 }
 for (int i=0;i<100;i++)
 if (var[i]!=i+1) 
 printf("Data Race Present
");
 return 0;
}

File: data/deepseek_coder_base_6.7B/readable_format_HPEC/fortran/file106.f90
program DRB155_missingordered_orig_gpu_no
    use omp_lib
    implicit none

    integer :: var(100)
    integer :: i

    do i = 1, 100
        var(i) = 1
    end do

    !$omp target map(tofrom:var) device(0)
    !$omp parallel do ordered
    do i = 2, 100
        !$omp ordered
        var(i) = var(i-1)+1
        !$omp end ordered
    end do
    !$omp end parallel do
    !$omp end target

    do i = 1, 100
        if (var(i)/=i) then
            print*,"Data Race Present"
        end if
    end do

end program
